attester/src/tdx/mod.rs

```diff
@@ -9,10 +9,24 @@ use serde::{Deserialize, Serialize};
 use std::path::Path;
 use tdx_attest_rs;
 
+mod az;
+
 const CCEL_PATH: &str = "/sys/firmware/acpi/tables/data/CCEL";
 
 pub fn detect_platform() -> bool {
-    Path::new("/dev/tdx-attest").exists() || Path::new("/dev/tdx-guest").exists()
+    // Upstream kernel exposes a /dev/tdx_guest device but Intel's lib expects
+    // /dev/tdx-guest
+    let paths = [
+        Path::new("/dev/tdx-attest"),
+        Path::new("/dev/tdx-guest"),
+        Path::new("/dev/tdx_guest"),
+    ];
```

Comment:
SGXDataCenterAttestationPrimitives would need to be fixed, because it uses a different ioctl/API to talk to /dev/tdx-guest than what was merged upstream. This also means symlinks won't simply work.

I think we're going to need to transition and phase out the older paths.